src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -761,9 +767,13 @@ static int parse_request(uint8_t code, size_t len, const void *option, void *use
 
                 break;
         case SD_DHCP_OPTION_RELAY_AGENT_INFORMATION:
-                req->agent_info_option = (uint8_t*)option - 2;
-
+                if (req->agent_info_option)
+                        mfree(req->agent_info_option);
```

Comment:
just use plain free() if you are not returned in the return value.

and also free()/mfree() happily accept NULL pointers and do nothing then

Reply:
> if you are not returned in the return value

I guess you meant "interested". Anyways, done.